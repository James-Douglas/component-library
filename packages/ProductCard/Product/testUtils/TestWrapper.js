import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
