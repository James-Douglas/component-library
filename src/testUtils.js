// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
