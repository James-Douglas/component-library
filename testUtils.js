import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
