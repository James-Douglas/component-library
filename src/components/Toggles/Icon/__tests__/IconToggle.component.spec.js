import React from 'react';
import { faCheck } from '@fortawesome/free-solid-svg-icons';
import { render, fireEvent } from '../../../../testUtils';
import IconToggle, { getToggleContent } from '../IconToggle.component';

describe('getToggleContent()', () => {
  const IconToggleContentContainer = ({
    // eslint-disable-next-line react/prop-types
    id, icon, title, description,
  }) => (
    <>
      {getToggleContent(id, icon, title, description)}
    </>
  );

  it('renders with props', () => {
    const { getByText, container } = render(<IconToggleContentContainer id="testing" icon={faCheck} title="test" description="test description" />);
    expect(getByText('test')).toBeInTheDocument();
    expect(getByText('test description')).toBeInTheDocument();
    expect(container.querySelector('.fa-check')).toBeInTheDocument();
  });
});

describe('IconToggle', () => {
  it('renders with props', () => {
    const { container } = render(<IconToggle id="test" value="test" title="test icon toggle" icon={faCheck} />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('calls handleToggle on toggle when provided', () => {
    const handleChangeCb = jest.fn();
    const { container } = render(<IconToggle id="test" value="test" title="test icon toggle" icon={faCheck} handleToggle={handleChangeCb} />);
    const element = container.firstChild;
    fireEvent.click(element);
    expect(handleChangeCb).toHaveBeenCalled();
    expect(handleChangeCb.mock.calls[0][0]).toEqual('test');
  });

  it('calls focus and blur handlers', () => {
    const handleFocus = jest.fn();
    const handleBlur = jest.fn();
    const { container } = render(<IconToggle id="test" title="testing" icon={faCheck} value="test" handleFocus={handleFocus} handleBlur={handleBlur} />);
    const input = container.querySelector('input');
    fireEvent.focus(input);
    expect(handleFocus).toHaveBeenCalled();
    fireEvent.blur(input);
    expect(handleBlur).toHaveBeenCalled();
  });
});
