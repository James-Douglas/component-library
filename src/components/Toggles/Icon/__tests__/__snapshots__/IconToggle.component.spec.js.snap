// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconToggle renders with props 1`] = `"<div class=\\"sc-AxjAm kBRMiC\\"><input tabindex=\\"0\\" id=\\"test\\" type=\\"radio\\" name=\\"\\" class=\\"sc-AxirZ cxuttB\\" value=\\"test\\"><label for=\\"test\\" class=\\"sc-AxiKw bMNIFU\\"><div class=\\"sc-AxhCb dEuFeY\\"><div class=\\"sc-AxhUy iozMHd\\"><div class=\\"sc-AxgMl hRVOuc\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"check\\" class=\\"svg-inline--fa fa-check fa-w-16 fa-4x \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z\\"></path></svg></div><div class=\\"sc-AxheI ilVCuK\\"><div class=\\"sc-Axmtr hZugSb\\">test icon toggle</div><div class=\\"sc-AxmLO hWpoUR\\"></div></div></div></div></label></div>"`;
